dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/AnonymisingDataService.java

```diff
@@ -0,0 +1,290 @@
+package uk.gov.justice.laa.crime.dces.integration.service;
+
+import lombok.extern.slf4j.Slf4j;
+import net.datafaker.Faker;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import uk.gov.justice.laa.crime.dces.integration.model.generated.contributions.CONTRIBUTIONS;
+
+import java.nio.ByteBuffer;
+import java.security.SecureRandom;
+import java.util.Optional;
+
+import static uk.gov.justice.laa.crime.dces.integration.utils.DateConvertor.convertToXMLGregorianCalendar;
+
+@Slf4j
+@Service
+public class AnonymisingDataService {
+
+    private static final long DEFAULT_SEED_VALUE = 38;
+    private Faker faker;
+
+    public CONTRIBUTIONS anonymise(CONTRIBUTIONS contribution) {
+
+        long seedId = DEFAULT_SEED_VALUE * contribution.getMaatId();
+        log.info("Anonymising data for contribution with maatId {} and seedId: {}", contribution.getMaatId(), seedId);
+
+        SecureRandom secureRandom = new SecureRandom(longToBytes(seedId));
```

Comment:
The swap over to SecureRandom will actually cause this to be non-deterministic, even with the same seed. This is due to the way the seed is used to generate the values using the default algorithm. You can see it if you run a test a few times, each time the first value will be different. 

You can just specify the algorithm to use, and that should fix it. Something like the below:
```
SecureRandom secureRandom = SecureRandom.getInstance("SHA1PRNG");
secureRandom.setSeed(seedId);
```

Tiny note that if we version change, there could potentially be a change in the implementation which will cause issues for this. But this isn't a long term solution, so should be all good.